src/components/common/DateTime/index.tsx

```diff
@@ -1,20 +1,25 @@
 import { ReactElement } from 'react'
 import { Tooltip } from '@mui/material'
-import { formatDateTime, formatTimeInWords } from '@/utils/date'
-import { isToday, startOfDay } from 'date-fns'
+import { formatDateTime, formatTime, formatTimeInWords } from '@/utils/date'
+import { useRouter } from 'next/router'
+import { AppRoutes } from '@/config/routes'
+import { useTxFilter } from '@/utils/tx-history-filter'
 
 const DAYS_THRESHOLD = 60
 
-const getRelevantTimestamp = (value: number) => {
-  return isToday(new Date(value)) ? value : startOfDay(value).getTime()
-}
-
 const DateTime = ({ value }: { value: number }): ReactElement => {
-  const displayExactDate = Math.floor((Date.now() - value) / 1000 / 60 / 60 / 24) > DAYS_THRESHOLD
+  const [filter] = useTxFilter()
+
+  const router = useRouter()
+  const isHistory = router.pathname === AppRoutes.transactions.history
+
+  const isOld = Math.floor((Date.now() - value) / 1000 / 60 / 60 / 24) > DAYS_THRESHOLD
+
+  const displayExactDate = isOld || isHistory
 
   return (
     <Tooltip title={displayExactDate ? '' : formatDateTime(value)} placement="top">
-      <span>{displayExactDate ? formatDateTime(value) : formatTimeInWords(getRelevantTimestamp(value))}</span>
+      <span>{filter ? formatDateTime(value) : displayExactDate ? formatTime(value) : formatTimeInWords(value)}</span>
```

Comment:
What about the queue?

Reply:
As per https://github.com/safe-global/web-core/issues/673, the queue will use "textual" time up until the threshold.

Reply:
The queue and filter have no date labels.

- As per #673, the queue should show "n days ago" until the `DAYS_THRESHOLD` is met and the history show "Jan 1, 1970 - 00:00:00 AM".
- However, as per your suggestion, the history should use "00:00:00 AM" as it has date labels and the filter use "Jan 1, 1970 - 00:00:00 AM" as it doesn't.

Reply:
Yes. From your code however, I see the following logic:
* If filter, show full date-time
* If history or over 60 days, show time when over 60 days
* If queue, it will show **only time** when over 60 days, otherwise "N days ago"